llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8761,20 +8762,23 @@ bool VPRecipeBuilder::getScaledReductions(
     return false;
 
   using namespace llvm::PatternMatch;
+  BinaryOperator *ExtendedBinOp = BinOp;
+  match(BinOp, m_Neg(m_BinOp(ExtendedBinOp)));
```

Comment:
Could this just set `BinOp` inside the match instead of creating a new variable?